lucille-plugins/lucille-pinecone/src/main/java/com/kmwllc/lucille/pinecone/util/PineconeManager.java

```diff
@@ -0,0 +1,34 @@
+package com.kmwllc.lucille.pinecone.util;
+
+import io.pinecone.clients.Pinecone;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import org.openapitools.control.client.model.IndexModelStatus.StateEnum;
+
+public class PineconeManager {
+  private static final String DEFAULT_NAMESPACE = "default";
+  private static final Map<String, Pinecone> clients = new ConcurrentHashMap<>();
```

Comment:
when might we need or want to close these clients?

one scenario is if the lucille config includes multiple connectors/pipelines. the first connector might be pinecone, the second might be something not relating to pinecone. Do we want to leave the pinecone clients open while the second connector is executing?

Reply:
Discussed this on call. Currently, the only places where a Pinecone client is used are stages & Indexers, both have methods that are called after it is done executing (stop() for stage, closeConnection() for Indexer). So in both cases where we have created a client, we have ending methods to close the connection. The Pinecone Client class has a package private closeConnection method, and only called when the Pinecone Index class calls close(). So to make sure that we are not leaving pinecone clients open unnecessary, both Stage and Indexer will call index.close() in their closing methods.

Reply:
After discussion, found out that pinecone clients are not used for connectors, only for stages and indexers. Brought up a good question in closing clients appropriately and edge cases like concurrent lucille runs in the same JVM. In the end, decided it is safer to not share clients and have each stage and indexer have their own instances of their connection, such that they could close the connection in their respective end methods (stop() in stages and closeConnection() in indexer).